storage/cache/log_subtree_cache_test.go

```diff
@@ -0,0 +1,116 @@
+package cache
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/google/trillian/merkle/rfc6962"
+	"github.com/google/trillian/storage"
+	"github.com/google/trillian/storage/storagepb"
+	"github.com/kylelemons/godebug/pretty"
+)
+
+const goldenSubtreeNoIntermediatesB64 = "H4sIAAAAAAAA/1xYvW7kRhPEB1zw4WwDhmznBz0BZzj8CxTscrnLf1KBAoUHwZYBBQoU6O0Nw1u1sKvSxrCnu7q6uoc////+169f2vb14e7H+ffvf3zrv3/8+S1l9799/dIO3cPdT/9aQ2yuhw+ZHA45XDyJNa+uLi7v6iIrr+bezDHiwkc15zDPb2quC5jNSV1eAzlpIEWOw+a6RoqrmmOI97/8k+LTw90P//GBDA+aYYT1Ua0IolNAK5w9a2jM4/JsGCHrZVFzg9Obno55dnV9TBoG0tu1gjEFxNFpHAFJd5I0jY9iTLjsRb6PZMCzhlBfQxifNI3UXM2TFa9CwL1RLqB6Y1JzAd+bpRcB0f5kWaN+naJfIpD1xRjAw5piReCU+g2iu1iKWUCKn5Yi2fxiTAR6pw+lLXqt07JW9dV6GTQMIm33Fei15WA8Ir0Maeay1gYpzPuHmgvg0WkgJW4cBmtjUPSsnVIngPdh4EE4euvXANejNUXC6dEwLVCBLVmvAKj905IBIrulXrDhzJyjMhftoYZhW+oBDdNqcRNAnawyJTXJWJ2RqMZqtsZqenC7sdYbK5BM61XgwtGCThRSJXvJkmsqFUCazEUJSCfr2wo1314NUpze3szM9jIngYqnGeYI76BWdnP7rmhQjlWka7DjUtt14NJiXMoA0lERDWxQHRUNyZsZS8GZLbO0UatOYyZ0g0VHKu3GglsJF603MDqpthVUPEWuAusu1lkZa2KpkKOjMToh6PVg2AHpwQqbSBpThAoFmM13BVBnO10D1NkCrBHgYszL0AEX2zAysHoxEWo43LW6JZLsDamAG3uLj4I62PyjNM2m4eTk8m5hg8An1SCO1t6AChR8k3BWcjMx5AJzVLGOlHBt/gaV6Q29SPW1C3Pw7KicDJDTow7nCKYetQciWa2x3eR7sRbAXD0r2UmP0XYuKuFsQ7HC6X0w11z9VCC5ZpzeNDp080WzblCqwcZZDuQG4wxHzkHRD9xJdEVoEva2g+xtGYyvYoRWHXVti6SWei1h7ZRwJRfYV+seNMRkaxFPT8aW2x5mbCEv2k/FmK6tTiU12riccV3VqAMq0mpFyM5Oa0oCDNaWfCCNljiZuJrIcHfZTAUjh6EtoOTAbi+TxLq+SQnZPEqMyGGvhS0oUzZAIjO0qvD9NpkCUusmax8ufouZuZjOduVNNgxU4jGYfEVut9YT3M4WWzwaThzr2ICCbWamoHSKVEWxU/Fp4GIxmWcn90aFyPeMjbhIoGzkc96stmOwFw/ai9zwzjqv+TycTVEqMkRXnUgXmneN4C5WF07rzQjJuuw2mqikk83fis9wg4mPgMV2bD4CVpP0m+KpJvBBMxj5crq2pZKPs8226Ygy7laBggqimh6p9LoHJAJi/czVu9VFMYfjkxax5HZlXcctZbSdgR09G8dI1NXEkCP8qOBFDIGT8qPgoNUO4Kt7tHYu+KK30pbw3Cp0tz8QauV2NlszVzU071k0j49Mmy58LOzWWUUOFy/igiXRuUAPk3UQa7LYfQ18nLWsNbjYm4+A5hxtbCX+r7DBcJsuJskFfxRoAUuKlU6zmv9NjEik6GQaVrLdrFVCBewGwY5/MazbmPhuLhI3FyUSX3CtrmbcclpVu4RyryYzgdRYJDYg1ClCfCj0pkj8WTQZm/mgna3cNWXDwqDI9HY6cKjalQlOVuPobdlVOQmcTdop1NHBKpgj6sF68PY3UsnBcm+mrlwbDopewAw66l7EX8GdbuJs+bMmwl+tvTEmUC5tWUp19tf//g4AAP//V1xHUZcWAAA="
```

Comment:
Might have put these in files in `testdata` but possibly hassle. Could format them as multi line strings though.